internal/admission/validator.go

```diff
@@ -532,14 +532,26 @@ func (validator KongHTTPValidator) ValidateVault(ctx context.Context, k8sKongVau
 	if err != nil {
 		return false, fmt.Sprintf(ErrTextVaultConfigUnmarshalFailed, err), nil
 	}
+
+	// list existing KongVaults and reject if the spec.prefix is duplicate with another `KongVault`.
+	existingKongVaults := validator.Storer.ListKongVaults()
+	prefixToKongVaultName := lo.SliceToMap(existingKongVaults, func(v *kongv1alpha1.KongVault) (string, string) {
+		return v.Spec.Prefix, v.Name
+	})
+	vaultName, ok := prefixToKongVaultName[k8sKongVault.Spec.Prefix]
```

Comment:
`prefixToKongVaultName` is used only once for the map lookup in 541. Because of that we can do away with the slice to map conversion and just use either a for loop or something like [`lo.Find`](https://pkg.go.dev/github.com/samber/lo@v1.39.0#Find) to find the element in the slice. This will retain the search time to be linear with respect to the number of elements but will get rid of the map allocation.

Reply:
replaced with `lo.Find`.